tldw_Server_API/app/core/AuthNZ/federation/oidc_service.py

```
@@ -0,0 +1,345 @@
+    """Construct OIDC authorization requests for trusted identity providers."""
+
+    default_scopes: tuple[str, ...] = ("openid", "email", "profile")
+    state_ttl_seconds: int = 600
+
+    async def _fetch_discovery_document(self, provider: dict[str, Any]) -> dict[str, Any]:
+        discovery_url = _coerce_nonempty_string(provider.get("discovery_url"))
+        if not discovery_url:
+            return {}
+        discovery = await afetch_json(method="GET", url=discovery_url)
+        if not isinstance(discovery, dict):
+            raise ValueError("OIDC discovery response is invalid")
```

Ruff (0.15.6)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)